src/System.Security.Cryptography.X509Certificates/tests/TestEnvironmentConfiguration.Unix.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    internal static partial class TestEnvironmentConfiguration
+    {
+        static partial void DetermineCanModifyStores(ref bool canModify)
+        {
+            canModify = DetermineCanModifyStores();
+        }
+
+        private static bool DetermineCanModifyStores()
+        {
+            uint userId = Interop.Sys.GetEUid();
+            string certStoresFeaturePath = PersistedFiles.GetUserFeatureDirectory("cryptography", "x509stores");
+
+            Directory.CreateDirectory(certStoresFeaturePath);
+
+            // Check directory permissions:
+
+            Interop.Sys.FileStatus dirStat;
+            if (Interop.Sys.Stat(certStoresFeaturePath, out dirStat) != 0)
+            {
+                return false;
+            }
+
+            if (dirStat.Uid != userId)
+            {
+                return false;
+            }
+
+            if ((dirStat.Mode & (int)Interop.Sys.Permissions.S_IRWXU) != (int)Interop.Sys.Permissions.S_IRWXU)
+            {
+                return false;
+            }
+
+            string probeFilename = Path.Combine(certStoresFeaturePath, "probe.chmod");
+
+            try
+            {
+                using (FileStream stream = new FileStream(probeFilename, FileMode.Create))
+                {
+                    Interop.Sys.FileStatus stat;
+                    if (Interop.Sys.FStat(stream.SafeFileHandle, out stat) != 0)
+                    {
+                        return false;
+                    }
+
+                    if (stat.Uid != userId)
+                    {
+                        return false;
+                    }
+
+                    // The product code here has a lot of stuff it does.
+                    // This capabilities probe will just check that chmod works.
+                    int mode = stat.Mode;
+
+                    // Flip all of the O bits.
+                    mode ^= (int)Interop.Sys.Permissions.S_IRWXO;
+
+                    if (Interop.Sys.FChMod(stream.SafeFileHandle, mode) < 0)
```

Comment:
Dunno, I copied it from the product code.  I'm assuming I knew what I was doing at the time :)